lib/tools/misc/arrayTools.js

```diff
@@ -0,0 +1,76 @@
+define([],
+function() {
+    "use strict";
+
+    function calcBounds(points){
```

Comment:
Most importantly, you are missing the length check of points at the beginning, instead you just expect there to be a `points[0]`.

Then, to stay more true to the original source, and to be more brief, this can be implemented like:

```js
function calcBounds(points){
    if(!points.length)
        return [0, 0, 0, 0];
    var xs = []
      , ys=[]
      , i
      ;
    for(i=0;i<points.length;i++){
        xs.push(points[i][0];
        ys.push(points[i][1];
    }
    return [Math.min.apply(null, xs), Math.min.apply(null, ys)
          , Math.max.apply(null, xs), Math.max.apply(null, ys)];
}

```